### PR TITLE
Lesson hw 2

### PR DESCRIPTION
**Pull request by kireeva67**

### CODE REVIEW

File: hw/hw2/task1.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+for(let ch = 0; ch < bulbsNumbers.length; ch++){
+    for(let ch1 = 0; ch1 < bulbsArray.length; ch1++){
+        if([ch1+1] % bulbsNumbers[ch] === 0){
+            bulbsArray[ch1] = (bulbsArray[ch1])? false:true;
+        }
+    }
+}
+
+let turnOnBulbs = bulbsArray.filter((x) => {return x === true});
```
Comment (alexmaltsv) [inline hw/hw2/task1.js]: bulbsArray.filter((x) => {return x === true}); -> bulbsArray.filter((x) => Boolean(x));

File: hw/hw2/task1.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+let number = 20;
+let bulbsNumbers = [2, 3, 8];
+let array = Array(number);
+let bulbsArray = Array.from(array,(x) => {return false});
+
+for(let ch = 0; ch < bulbsNumbers.length; ch++){
```
Comment (alexmaltsv) [inline hw/hw2/task1.js]: What mean ch? Provide better name

File: hw/hw2/task1.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
+let number = 20;
```
Comment (alexmaltsv) [inline hw/hw2/task1.js]: Better to make a function and all the variables in it 

so, should be like calcBulbsAmount and all this code in that  

This applies to all tasks

File: hw/hw2/task4.js [hunk 4]
```diff
@@ -0,0 +1,63 @@
+let number = 3;
+
+function createMagicSquare(number){
+    let array =  Array(number);
+    let magicSquare = Array.from(array,(x) => {return 0});
+    magicSquare = magicSquare.map((x) => {return x = Array.from(Array(number),(x) => {return 0})});
```
Comment (alexmaltsv) [inline hw/hw2/task4.js]: - Don't use comma, it's hard to read 
- Don't use return and assignment in one line 

- (x) => {return 0} should be x => 0

File: hw/hw2/task4.js [hunk 5]
```diff
@@ -0,0 +1,63 @@
+let number = 3;
+
+function createMagicSquare(number){
+    let array =  Array(number);
+    let magicSquare = Array.from(array,(x) => {return 0});
```
Comment (alexmaltsv) [inline hw/hw2/task4.js]: Better, instead, Array.from(array,(x) => {return 0}); => `array.fill(0)`

File: hw/hw2/task1.js (hunk 1 above)
Comment (kireeva67) [reply hw/hw2/task1.js]: fixed

File: hw/hw2/task1.js (hunk 2 above)
Comment (kireeva67) [reply hw/hw2/task1.js]: fixed

File: hw/hw2/task1.js (hunk 3 above)
Comment (kireeva67) [reply hw/hw2/task1.js]: fixed

File: hw/hw2/task4.js (hunk 4 above)
Comment (kireeva67) [reply hw/hw2/task4.js]: fixed

File: hw/hw2/task4.js (hunk 5 above)
Comment (kireeva67) [reply hw/hw2/task4.js]: fixed

**Review by alexmaltsv — APPROVED** [review]

**Pull request merged**